src/ducks/connections/index.js

```diff
@@ -280,66 +382,113 @@ export const getConnectionStatus = (
+  if (trigger.isConnected) return 'connected'
+  return 'errored'
+}
+
+export const getQueue = (state, registryKonnectors) =>
   // state is state.connections
-  return Object.keys(state).reduce((queuedConnections, konnectorSlug) => {
-    const triggers = state[konnectorSlug].triggers
-    if (!triggers) return queuedConnections
-    const registryKonnector = registryKonnectors.data[konnectorSlug]
-    return queuedConnections.concat(
-      Object.keys(triggers).reduce((queuedTriggers, triggerId) => {
-        if (triggers[triggerId].isEnqueued) {
-          const label = registryKonnector.name
-          const status = getConnectionStatusFromTrigger(triggers[triggerId])
-          const icon = getKonnectorIcon(registryKonnector)
-          return queuedTriggers.concat({ label, status, icon })
-        }
+  state.konnectors
+    ? Object.keys(
+        state.konnectors
+      ).reduce((queuedConnections, konnectorSlug) => {
+        const triggers = state.konnectors[konnectorSlug].triggers
+        if (!triggers) return queuedConnections
+        const registryKonnector = registryKonnectors.data[konnectorSlug]
+        return queuedConnections.concat(
+          Object.keys(triggers).reduce((queuedTriggers, triggerId) => {
+            if (triggers[triggerId].isEnqueued) {
+              const label = registryKonnector.name
+              const status = getTriggerQueueStatus(triggers[triggerId])
+              const icon = getKonnectorIcon(registryKonnector)
+              return queuedTriggers.concat({ label, status, icon })
+            }
 
-        return queuedTriggers
+            return queuedTriggers
+          }, [])
+        )
       }, [])
-    )
-  }, [])
-}
+    : []
 
 export const getConfiguredKonnectors = (state, existingAccountIds = []) =>
-  Object.keys(state).filter(
-    konnectorSlug =>
-      !!state[konnectorSlug] &&
-      !!Object.keys(state[konnectorSlug].triggers).find(triggerId => {
-        const connection = state[konnectorSlug].triggers[triggerId]
-        return (
-          existingAccountIds.includes(connection.account) &&
-          (connection.isConnected ||
-            connection.isRunning ||
-            connection.isEnqueued ||
-            connection.hasError)
-        )
-      })
-  )
+  state.konnectors
+    ? Object.keys(state.konnectors).filter(
+        konnectorSlug =>
+          !!state.konnectors &&
+          !!state.konnectors[konnectorSlug] &&
+          !!Object.keys(
+            state.konnectors.konnectors[konnectorSlug].triggers
+          ).find(triggerId => {
+            const connection =
+              state.konnectors[konnectorSlug].triggers[triggerId]
+            return (
+              existingAccountIds.includes(connection.account) &&
+              (connection.isConnected ||
+                connection.isRunning ||
+                connection.isEnqueued ||
+                connection.hasError)
+            )
+          })
+      )
+    : []
```

Comment:
Kind of the same here, it's a `.keys.filter.keys.find` :|

Reply:
I kind of LOVE one line instructions.

Reply:
I will keep thoses unclear functions for now. I think the better way is not to use more line but to split every `keys`/`filter`/`find` in more explicit functions. It needs a little focus, work and especially test so I will improve this part in a dedicated PR, hopefully next week. Thanks !